Objects/typeobject.c

```diff
@@ -8064,7 +8064,7 @@ static slotdef slotdefs[] = {
     UNSLOT("__abs__", nb_absolute, slot_nb_absolute, wrap_unaryfunc,
            "abs(self)"),
     UNSLOT("__bool__", nb_bool, slot_nb_bool, wrap_inquirypred,
-           "self != 0"),
+           "True if self else False"),
```

Comment:
This is incorrect in general (e.g. for range objects).

Reply:
Rather than propose adding the rule for ranges (I believe 'self.start != self.stop') and unknown others present or future, I withdraw my proposal for a non-circular rule.